### PR TITLE
adding scheduled events to audit log

### PR DESCRIPTION
**Pull request by SaculRennorb**
I'll be frank, I just made this because i needed the `user` and `affected_id` for ScheduledEvent updates, as the corresponding event does not provide a field of who updated the ScheduledEvent. 

Was sad to see that apparently noone added auditLog ids 100-102 (the ones related to ScheduledEvents) and there does not seen to be a generic way of just getting an arbitrary field from those log entries (can't even really do reflection magic because of how the data is set up).

This isn't tested very well and might also still contain some copy paste problems, as well as problems related to me not knowing the inner workings of discord.net, also I didn't bother too much with getting used to style / structures of discord.net. Feel free to reject this if you don't want to put in work, as neither do I. It seems to work, but also I just wanted to have the ids.

### CODE REVIEW

File: src/Discord.Net.Rest/Entities/AuditLogs/DataTypes/ScheduledEventUpdateAuditLogData.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            return new ScheduledEventUpdateAuditLogData(id, before, after);
+        }
+
+        // Doc Note: Corresponds to the *current* data
+
+        /// <summary>
+        ///     Gets the snowflake id of the event.
+        /// </summary>
+        public ulong Id { get; }
+        public ScheduledEventInfo Before { get; }
```
Comment (quinchs) [inline src/Discord.Net.Rest/Entities/AuditLogs/DataTypes/ScheduledEventUpdateAuditLogData.cs:93]: Missing docnote

File: src/Discord.Net.Rest/Entities/AuditLogs/DataTypes/ScheduledEventUpdateAuditLogData.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        public ScheduledEventInfo After { get; }
```
Comment (quinchs) [inline src/Discord.Net.Rest/Entities/AuditLogs/DataTypes/ScheduledEventUpdateAuditLogData.cs:97]: Missing docnote

**Review by quinchs — CHANGES_REQUESTED** [review]

File: src/Discord.Net.Rest/Entities/AuditLogs/DataTypes/ScheduledEventUpdateAuditLogData.cs (hunk 1 above)
Comment (SaculRennorb) [reply src/Discord.Net.Rest/Entities/AuditLogs/DataTypes/ScheduledEventUpdateAuditLogData.cs:93]: added a doc text

File: src/Discord.Net.Rest/Entities/AuditLogs/DataTypes/ScheduledEventUpdateAuditLogData.cs (hunk 2 above)
Comment (SaculRennorb) [reply src/Discord.Net.Rest/Entities/AuditLogs/DataTypes/ScheduledEventUpdateAuditLogData.cs:97]: added a doc text

**Review by quinchs — APPROVED** [review]

**Comment by quinchs** [comment]
@SaculRennorb please update from dev

**Pull request merged**